### ISSUE TITLE
Fix bug with "Download Sensor Data" button

### ISSUE THREAD

**Issue by aufdenkampe** [ODM2/ODM2DataSharingPortal#304]
On production, @fisherba and I couldn't get the "Download Sensor Data" button to download a file for all the variables at a site. Downloading individual variables still worked.

**Comment by jcaraballo17** [comment]
Just tried downloading the data for 5 different sites on production and it worked.

@fisherba can you give me more specific information on this? 
I can figure out if that specific site you tried is having problems with the data, which site didn't work?
It could also be a browser-specific issue, what browser are you using?

**Comment by Maurier** [comment]
It looks like the method we are using to provide the download is not supported in Safari yet.
See https://caniuse.com/#feat=download

**Comment by fisherba** [comment]
The download of combined data worked today on the following sites: 
http://data.envirodiy.org/sites/NLP01/
http://data.envirodiy.org/sites/NLP02/
http://data.envirodiy.org/sites/CMP01/
http://data.envirodiy.org/sites/CMP02/
so I thought it was maybe a fluke that it didn't work on October 3 & 4. But I went back to the site that gave me trouble and it did not work today. 
http://data.envirodiy.org/sites/BIXBY_7/ is the site that won't allow me to download as combined csv (individual monitored values download). 
When I try to download at this site, instead of the "SiteName_TimeSeriesResults", the browser says "download" and then Chrome declares "Failed - Server problem". 

Chrome Version 68.0.3440.106 and Mac OS 10.13.6

**Comment by horsburgh** [comment]
Thanks for doing some quick testing @fisherba. @jcaraballo17 - can you look at the site that is still having issues and see if you can figure out what's up with that one?

**Comment by aufdenkampe** [comment]
@fisherba and I just found a major bug with this download csv feature -- 
***Data from the different sensors are not aligned properly by row according to their timestamps!***

She writes:
> I downloaded data for Williams Street Pond from data.envirodiy. Some of the variables began logging in July. I added redox, DO, and conductivity on September 26. The spreadsheet has all variables beginning in July (with redox, DO, and conductivity having no data after July 30). In other words, the CSV download put all values at the top of the spreadsheet.

She is specifically referring to these sites:
* WSP01: https://data.envirodiy.org/sites/WSP01/ and it's downloaded [WSP01_TimeSeriesResults.csv](https://drive.google.com/open?id=14YoCEW6gqVRpRBSuyugyokH0av_-pPNj) file.
* WSP02: https://data.envirodiy.org/sites/WSP02/ and it's downloaded [WSP02_TimeSeriesResults.csv](https://drive.google.com/open?id=1TsZD3zHx6DscD8miBlHnxxGMpKbNQRei) file.

You'll note the additional issue that any blank values for sensors are not given their own row, so that these also cause data misalignment. This happens relatively frequently. To see this, scroll down the bottom of the spreadsheets to see that there are often a slightly different number of rows for different variables, even for sensors that have been deployed at the same time and should have exactly the same number of rows.

Data from the newer sensors-variables have proper timestamps when downloaded individually.
This suggests that the issue is about data alignment when merging the datasets.

@jcaraballo17, I strongly urge the use of the Pandas Python library for these data merging and manipulation tasks, as Pandas functions auto-align data with no errors and also do it very quickly.

I also urge the use of Pandas with CSV upload functions, as suggested in https://github.com/ODM2/ODM2DataSharingPortal/issues/294#issuecomment-429695024, would also assist with parsing a much wider range of datetime formats and delimiters, which would alleviate the many Excel-introduced issues described in https://github.com/ODM2/ODM2DataSharingPortal/issues/12#issuecomment-413281393.

**Comment by horsburgh** [comment]
@darksinge worked on this. He and @jcaraballo17 will have a look and see what's up.

**Comment by horsburgh** [comment]
I tested this one on the sandbox using this site: http://envirodiysandbox.usu.edu/sites/BIXBY_7/

The other sites listed in @aufdenkampe 's comment above don't exist in the sandbox database. Download of the file containing all variables for the Bixby site worked (where it didn't before), and the combined data file had appropriate null values for where data do not exist. I checked several of the time series in the file to make sure things were beginning where they were supposed to and ending where they were supposed to. I compared the download file for individual variables to the download file for all variables in one file, and the data all seemed organized correctly.

@aufdenkampe, @SRGDamia1, and @fisherba - if there are other sites you know of where this was a problem, it might be good to have a couple of other independent tests. I'm going to add the "tested" label for now, but if you find any additional issues, we can revisit this one.
